twitter_analysis/app.py

```diff
@@ -0,0 +1,166 @@
+            verified = user['verified']
+            geo_enabled = user['geo_enabled']
+            default_profile = user['default_profile']
+            default_profile_image = user['default_profile_image']
+            favourites_count = user['favourites_count']
+            followers_count = user['followers_count']
+            friends_count = user['friends_count']
+            statuses_count = user['statuses_count']
+            average_tweets_per_day = np.round(statuses_count / 1+account_age_days, 3)
+
+            # manufactured features
+            hour_created = int(datetime.strptime(user['created_at'],'%a %b %d %X %z %Y').replace(tzinfo=None).strftime('%H'))
+            network = np.round(np.log(1 + friends_count)
+                            * np.log(1 + followers_count), 3)
+            tweet_to_followers = np.round(
+                np.log(1 + statuses_count) * np.log(1 + followers_count), 3)
+            follower_acq_rate = np.round(
+                np.log(1 + (followers_count / 1+account_age_days)), 3)
+            friends_acq_rate = np.round(
+                np.log(1 + (friends_count / 1+account_age_days)), 3)
+
+            # organizing list to be returned
+            user_features = [verified, hour_created, geo_enabled, default_profile, default_profile_image,
+                                favourites_count, followers_count, friends_count, statuses_count,
+                                average_tweets_per_day, network, tweet_to_followers, follower_acq_rate,
+                                friends_acq_rate]
+            # features for model
+            features = ['verified', 'hour_created', 'geo_enabled', 'default_profile', 'default_profile_image',
+                        'favourites_count', 'followers_count', 'friends_count', 'statuses_count', 'average_tweets_per_day',
+                        'network', 'tweet_to_followers', 'follower_acq_rate', 'friends_acq_rate']
+
+            # creates df for model.predict() format
+            user_df = pd.DataFrame(np.matrix(user_features), columns=features)
+            xgb_model = self.load_pkl('model_file_name.json')
+            prediction = xgb_model.predict(user_df)[0]
+            f = open('bot_data.csv', 'a')
+            writer = csv.writer(f)
+            user_features.append(prediction)
+            # write a row to the csv file
+            writer.writerow(user_features)
+
+            # close the file
+            f.close()
+            
+            if prediction==0:
+                print('Added')
+                collection.insert_one({'tweet':i['full_text'],'language':i['lang'],'created_at':str(datetime.strptime(i['created_at'],'%a %b %d %X %z %Y').replace(tzinfo=None).date()),'topic':q_obj['topic'],'query':q_obj['query']})
+            
+        print('Number of tweets: '+str(count))
+    
+    def load_pkl(self,fname):
+        model2 = XGBClassifier()
+        model2.load_model(fname)
+        return model2
+    
+    def clean_data(self):
```

Comment:
either add more descriptive name or place these functions inside specific classes. eg, this would now be tweedmind.tweettools.clean_date(self)